src/App.js

```diff
@@ -1,23 +1,74 @@
-import logo from './logo.svg';
+import contactsArr from './contacts.json'
+import {useState} from 'react'
 import './App.css';
 
+
+
 function App() {
+  let fiveContacts =  [...contactsArr.slice(0, 5)]
+  
+  let [contactList, setContacts] = useState(fiveContacts)
+
+  const addContact = ( () => {
+    let random = Math.floor( (Math.random() * contactsArr.length) -1)
+    
+    setContacts( (contactList)=>{
+      const newArr = contactsArr.filter((el, index) => {
+        if (index === random){
+          return true
+        } else {
+          return false
+        }
+      })
+      return contactList.concat(newArr)
+    })
+
+  })
+  
+  const sortByName = ( () => {
+    const orderName = [...contactList.sort((a, b) => a.name.localeCompare(b.name))]
+    setContacts(orderName)
+  })
+  const sortByPop = ( () => {
+    const orderName = [...contactList.sort((a, b) => a.popularity - b.popularity)]
+    setContacts(orderName)
+  })
+
+  const deleteRow = ((el) => {
+    const target = document.getElementById(el.target.id)
+    target.parentNode.removeChild(target)
+    setContacts(contactList)
+  }) 
```

Comment:
react is doing something like dom manipulation, so we should avoid to use dom manipulation with react.
what we do instead is passing the contact id to the function inside our map method. So when we map over contacts, we display contact.name and all that stuff, so we can use contact.id (if you check json data, we have an id) and we pass that id to the delete function as an argument and our function can look like this:
```
 const deleteContact = (contactId) =>{
    setContacts((prevContacts)=>{
      const newContactsList = prevContacts.filter((contact)=> {
        return contact.id !== contactId;
      });
      return newContactsList
    })
  }
```

inside return/ inside map() :
    `            <td> <button onClick={ () => deleteContact(contact.id) }>Delete</button> </td>`